src/MendelIHT.jl

```diff
@@ -109,25 +213,35 @@ function L0_reg(
     # initialize booleans
     converged = false             # scaled_norm < tol?
 
-    #convert bitarrays to Float64 genotype matrix, standardize each SNP, and add intercept
-    snpmatrix = convert(Array{Float64,2}, x.snpmatrix)
-    # snpmatrix = use_A2_as_minor_allele(x.snpmatrix) #to compare results with using PLINK
-    snpmatrix = StatsBase.zscore(snpmatrix, 1) 
-    snpmatrix = [ones(size(snpmatrix, 1)) snpmatrix] 
+    # compute some summary statistics for our snpmatrix
+    maf, minor_allele, missings_per_snp, missings_per_person = summarize(x)
+    people, snps = size(x)
+
+    #precompute mean and standard deviations for each snp. Note that (1) the mean is 
+    #given by 2 * maf, and (2) based on which allele is the minor allele, might need to do 
+    #2.0 - the maf for the mean vector.
+    mean_vec = zeros(snps) 
+    for i in 1:snps
+        minor_allele[i] ? mean_vec[i] = 2.0 - 2.0maf[i] : mean_vec[i] = 2.0maf[i] 
+    end
+    std_vec = std_reciprocal(x, mean_vec)
+
+    #add intercept (at the end)
+    x        = [x SnpArray(ones(people))] #NOTE this creates A LOT of extra memory!!!!
```

Comment:
if you really need to add an intercept, but `SnpArray` input cannot handle it, then consider copying the phenotype into a new vector and subtracting off the mean. this centers the phenotype. it should be equivalent to regressing onto a vector of 1s but dramatically more memory-efficient than adding an intercept to a `SnpArray` ;-)

Reply:
sorry I guess I don't get it, can you elaborate more? Or if you know some reference?

Reply:
insetad of using ` x = [x SnpArray(ones(people))]`, can you leave `x` alone and insert at the beginning of your function something like

    y_copy = copy(y)
    y_copy .-= mean(y_copy)

and use `y_copy` instead of `y` in your function?

Reply:
Ok I think I get the rough idea, but it'll take me awhile to make it work, so I'll leave it for now.... I'll submit a new pr for the changes you mentioned